### PR TITLE
EASY-2714: integration of easy-deposit-properties in easy-deposit-api

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes EASY-2714

#### When applied it will
* do the integration of `easy-deposit-properties` in `easy-deposit-api`

See also the configuration in https://github.com/DANS-KNAW/easy-dtap/pull/473

@DANS-KNAW/easy for review

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala [hunk 1]
```diff
@@ -183,7 +214,7 @@ class EasyDepositApiApp(configuration: Configuration) extends DebugEnhancedLoggi
     trace(user)
     for {
       depositDir <- DepositDir.create(draftBase, user)
-      depositInfo <- depositDir.getDepositInfo(submitBase, easyHome)
+      depositInfo <- depositDir.getDepositInfo(depositPropertiesRepo, easyHome)
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala:217]: I though we would only use graphql for the submitted bags, the drafts would stick to files. These properties belong to the draft.

File: src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiService.scala [hunk 2]
```diff
@@ -25,27 +24,27 @@ import org.scalatra.servlet.ScalatraListener
...
-  info(s"HTTP port is ${ serverPort }")
+  private val server = new Server(serverPort) {
+    setHandler(
+      new ServletContextHandler(ServletContextHandler.NO_SESSIONS & ServletContextHandler.SECURITY) {
+        addEventListener(new ScalatraListener() {
+          override def probeForCycleClass(classLoader: ClassLoader): (String, LifeCycle) = {
+            ("anonymous", new LifeCycle {
+              override def init(context: ServletContext): Unit = {
+                for ((path, servlet) <- servlets)
+                  context.mount(servlet, path)
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiService.scala:39]: what happened with the "/*"?

File: src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala [hunk 3]
```diff
@@ -79,8 +79,8 @@ object Errors extends DebugEnhancedLogging {
   case class PropertyNotFoundException(key: String, props: PropertiesConfiguration)
     extends PropertyException(s"no value for '$key' in ${ Option(props.getFile).getOrElse("not found properties file") }")
 
-  case class InvalidPropertyException(key: String, value: String, props: PropertiesConfiguration)
-    extends PropertyException(s"Not expected value '$value' for '$key' in ${ props.getFile }")
+  case class InvalidPropertyException(key: String, value: String, id: UUID)
+    extends PropertyException(s"Not expected value '$value' for '$key' in $id")
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala:83]: You lost info whether the properties are about the draft or about the submitted bag.

File: src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala (hunk 1 above)
Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala:217]: You are right, but in retrieving the `DepositInfo`, you need the `StateManager`, which might be looking into the submitted deposit properties (if the draft state is `SUBMITTED` or `IN_REVIEW`), which requires the GraphQL service.

File: src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiService.scala (hunk 2 above)
Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiService.scala:39]: It went into `Command`: https://github.com/DANS-KNAW/easy-deposit-api/pull/219/files#diff-ac6344ec10b2f8329af08a2135471040R54-R60

**Pull request merged**